src/commands/react-native-ci-cli.ts

```diff
@@ -68,21 +76,33 @@ const runReactNativeCiCli = async (toolbox: CycliToolbox) => {
     'Created snapshot of project state before execution.'
   )
 
-  const lintExecutor = await lint.run(toolbox, context)
-  const jestExecutor = await jest.run(toolbox, context)
-  const typescriptExecutor = await typescriptCheck.run(toolbox, context)
-  const prettierExecutor = await prettierCheck.run(toolbox, context)
-  const easUpdateExecutor = await easUpdate.run(toolbox, context)
-  const detoxExecutor = await detox.run(toolbox, context)
-
-  const executors = [
-    lintExecutor,
-    jestExecutor,
-    typescriptExecutor,
-    prettierExecutor,
-    easUpdateExecutor,
-    detoxExecutor,
-  ].filter((executor) => executor != null)
+  const featureFlags = getFeatureOptions().map((option) => option.flag)
+
+  const selectedFeatureFlags = toolbox.options.isPreset()
+    ? intersection(featureFlags, Object.keys(toolbox.parameters.options))
+    : await toolbox.interactive.multiselect(
+        'Select workflows you want to run on every PR',
+        `Learn more about PR workflows: ${REPOSITORY_FEATURES_HELP_URL}`,
+        RECIPES.map((recipe: CycliRecipe) => ({
+          label: recipe.meta.name,
+          value: recipe.meta.flag,
+          hint: recipe.meta.selectHint,
+        }))
+      )
+
+  // EAS Update recipe is currently supported only for Expo projects
+  if (
```

Comment:
Hmm, I don't like that we check EAS Update specififc logic here, It's just one thing so it could be for now, but few thoughts: 
1. Could we gray out expo and in hint say that it's not supported in non expo project?
2. If more ifs like this will come, maybe we could add some method like `execute` for validation 
3. Maybe we could detect setup Expo/React Native/(React/Node in the future) and display what we detected and what is not compatible (potentialy add prompt to "skip checks" and generate just workflows on user risk) 
wdyt?

Reply:
Maybe something like this:

- I'll try to add option to print "disabled" (greyed out) options in multiselect (clack has no such thing, but I copied the code anyway so maybe it's feasible) and we could use it to print options that we detected as not supported/recommended for current project
- We can add `validate()` function to each recipe that receives context and throws error if the recipe is not supported
- in main function we can call `validate()` for each selected recipe (through `--preset`, in multiselect we don't have to, because we won't let to choose unsupported options) 
- in the future we can add a flag like you said, `--skip-checks` to skip all these steps

Reply:
Ok I added disabled options to multiselect and `validate()` to `CycliRecipe` type. Now EAS Update is disabled in non-expo projects. 9c1def7afe372e509ae91a2ab9f59e75535a1bd8

<img width="1069" alt="Zrzut ekranu 2024-08-30 o 10 52 27" src="https://github.com/user-attachments/assets/cd1e86c7-91ff-40ed-b7d1-42244a25d9df">